testsuite/tests/service_mesh/auth/wasm/test_mapping_rules_sync.py

```diff
@@ -0,0 +1,51 @@
+"""
+Tests wasm mapping rules special handling.
+"""
+import backoff
+import pytest
+from testsuite.capabilities import Capability
+from testsuite import rawobj
+from testsuite.utils import blame
+
+
+pytestmark = pytest.mark.required_capabilities(Capability.SERVICE_MESH)
+
+
+@pytest.fixture(scope="session")
+def set_mapping_rules(request):
+    """Deletes implicit '/' rule and sets new for path '/anything/something'"""
+    def _set_mapping_rules(service):
+        proxy = service.proxy.list()
+        proxy.mapping_rules.delete(proxy.mapping_rules.list()[0]["id"])  # delete implicit "/" rule
+        metric = service.metrics.create(rawobj.Metric(blame(request, "metric")))
+        proxy.mapping_rules.create(rawobj.Mapping(metric, pattern="/anything/something"))
+        proxy.deploy()
+    return _set_mapping_rules
+
+
+@pytest.fixture(scope="session")
+def backoff_request():
+    """
+    Get request which retries for max a minute if unexpected status code is returned.
+    Returns function that returns False if after backoff the expected status code is not returned.
+    """
+    @backoff.on_predicate(backoff.constant, lambda x: not x, interval=10, max_tries=6, jitter=None)
+    def _backoff_request(client, url, expected_return=200):
+        return client.get(url).status_code == expected_return
+
+    return _backoff_request
+
+
+def test_mapping_rules_automatic_sync(api_client, service, set_mapping_rules, backoff_request):
+    """
+    Wasm synchronizes mapping rules from 3scale periodically. Usually every 20-60s.
+    This test changes mapping rules in proxy config and waits for wasm to sync.
+    """
+    client = api_client(disable_retry_status_list=[404])
+
+    assert client.get("/anything/anything").status_code == 200
+
+    set_mapping_rules(service)
+
+    assert backoff_request(client, "/anything/anything", 404)  # waiting for synchronisation
```

Comment:
if necessary (a question above), maybe creating new client might be suitable (calling api_client again) and few lines would be made redundant.